addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/HPPrinterBindingConstants.java

```diff
@@ -0,0 +1,138 @@
+    public static final ThingTypeUID THING_TYPE_OFFICEJET_4650 = new ThingTypeUID(BINDING_ID, HP_TYPE_OJ4650);
+    public static final ThingTypeUID THING_TYPE_OFFICEJET_6600 = new ThingTypeUID(BINDING_ID, HP_TYPE_OJ6600);
+    public static final ThingTypeUID THING_TYPE_OFFICEJET_7500_e910 = new ThingTypeUID(BINDING_ID, HP_TYPE_OJ7500e910);
+
+    // OfficeJet Pros
+    public static final ThingTypeUID THING_TYPE_OFFICEJETPRO_X476DWMFP = new ThingTypeUID(BINDING_ID,
+            HP_TYPE_OJPX476DWMFP);
+
+    public static final ThingTypeUID THING_TYPE_OFFICEJETPRO_8600 = new ThingTypeUID(BINDING_ID, HP_TYPE_OJP8600);
+    public static final ThingTypeUID THING_TYPE_OFFICEJETPRO_8710 = new ThingTypeUID(BINDING_ID, HP_TYPE_OJP8710);
+    public static final ThingTypeUID THING_TYPE_OFFICEJETPRO_8730 = new ThingTypeUID(BINDING_ID, HP_TYPE_OJP8730);
+
+    // ENVYs
+    public static final ThingTypeUID THING_TYPE_ENVY_4520 = new ThingTypeUID(BINDING_ID, HP_TYPE_E4520);
+    public static final ThingTypeUID THING_TYPE_ENVYPHOTO_6200 = new ThingTypeUID(BINDING_ID, HP_TYPE_EP6200);
+
+    // DeskJets
+    public static final ThingTypeUID THING_TYPE_DESKJET_2620 = new ThingTypeUID(BINDING_ID, HP_TYPE_DJ2620);
+    public static final ThingTypeUID THING_TYPE_DESKJET_3520 = new ThingTypeUID(BINDING_ID, HP_TYPE_DJ3520);
+
+    // LaserJets
+    public static final ThingTypeUID THING_TYPE_COLORLASERJET_MFP277DW = new ThingTypeUID(BINDING_ID,
+            HP_TYPE_CLJMFP277DW);
+    public static final ThingTypeUID THING_TYPE_LASERJETPRO_200COLORM251NW = new ThingTypeUID(BINDING_ID,
+            HP_TYPE_LJP200CM251NW);
+    public static final ThingTypeUID THING_TYPE_LASERJETPRO_M14_M17 = new ThingTypeUID(BINDING_ID,
+            HP_TYPE_LJPM14_LJPM17);
+
+    public static final ThingTypeUID THING_TYPE_GENERIC1 = new ThingTypeUID(BINDING_ID, HP_TYPE_GENERIC1);
+
+    public static final Set<ThingTypeUID> SUPPORTED_DEVICES = Stream.of(THING_TYPE_OFFICEJET_3831,
+            THING_TYPE_OFFICEJET_4650, THING_TYPE_OFFICEJET_6600, THING_TYPE_OFFICEJET_7500_e910,
+            THING_TYPE_OFFICEJETPRO_X476DWMFP, THING_TYPE_OFFICEJETPRO_8600, THING_TYPE_OFFICEJETPRO_8710,
+            THING_TYPE_OFFICEJETPRO_8730, THING_TYPE_ENVY_4520, THING_TYPE_ENVYPHOTO_6200, THING_TYPE_DESKJET_2620,
+            THING_TYPE_DESKJET_3520, THING_TYPE_COLORLASERJET_MFP277DW, THING_TYPE_LASERJETPRO_200COLORM251NW,
+            THING_TYPE_LASERJETPRO_M14_M17, THING_TYPE_GENERIC1).collect(Collectors.toSet());
+
+    // UPNP Device Lookup Table. Normally UPNP device is looked up by Device ID, however for printers
+    // with large Model Names, they have to be looked up using the below table due to Device ID length
+    // restrictions. Will return Generic Device if unable to find supporting model.
+    public static ThingTypeUID upnpDeviceLookup(String name) {
+        final Hashtable<String, ThingTypeUID> dv = new Hashtable<String, ThingTypeUID>();

```

Comment:
Minor code style: You can use the diamond operator for creating the Hashtable.

```suggestion
	        final Hashtable<String, ThingTypeUID> dv = new Hashtable<>();
```